### PR TITLE
ci: add prettier and eslint checks in actions and fix nodejs actions

### PR DESCRIPTION
**Pull request by bengeois**
#### Checks 

This merge request adds :
- a lint check
- a Prettier check
- a fix to the documentation job to run exclusively on Node.js 20.

These improvements ensure consistent code style, prevent formatting issues, and streamline the CI workflow by limiting the documentation job to a specific Node.js version.

#### Triggers
Setting the workflow to trigger on every pull request and only on pushes to develop branch. This ensures that feature branches and fixes are tested before merging without redundant checks on other branches, focusing CI resources where they're needed most.

#### Unused workflow
Also delete merge-conflict-autolabel.yml as I don't think it was working. Also Github already indicates when MR has merge conflicts

### CODE REVIEW

File: .github/workflows/merge-conflict-autolabel.yml
Comment (VaiTon) [inline .github/workflows/merge-conflict-autolabel.yml:1]: Why did you remove this one?

**Comment by VaiTon** [comment]
@bengeois let's do something else. We keep this PR for eslint and prettier and I create a new one for the workflow

**Comment by VaiTon** [comment]
@bengeois please rebase on https://github.com/openfoodfacts/openfoodfacts-nodejs/pull/552

File: .github/workflows/merge-conflict-autolabel.yml
Comment (bengeois) [reply .github/workflows/merge-conflict-autolabel.yml:1]: I suggest this change for two reasons.

First this workflow has never run, you can see this in previous pull request or in the history of the Github Actions tab, it is configured on the branch named "main"
Second, gitHub’s interface already flags pull requests with merge conflicts directly.

This minimizes the need for an extra action dedicated to merge conflict detection.

Comment (VaiTon) [reply .github/workflows/merge-conflict-autolabel.yml:1]: @teolemon created it (if I remember correctly) so let's see what he thinks. I'm all for removing it

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+      # Run build
+      - name: Build project
+        run: yarn build
+
+      # Run tests
+      - name: Run tests
+        run: yarn test
+
+      # Build documentation only for Node.js 20.x
```
Comment (VaiTon) [inline .github/workflows/nodejs.yml]:
```suggestion
      # Build documentation only for Node.js 22.x
```

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Build documentation
+        if: matrix.node-version == '22.x'
+        run: yarn build:docs
+
+      # Upload documentation artifact (only for Node.js 20.x)
```
Comment (VaiTon) [inline .github/workflows/nodejs.yml]:
```suggestion
      # Upload documentation artifact (only for Node.js 22.x)
```

**Review by VaiTon — CHANGES_REQUESTED** [review]

File: .github/workflows/merge-conflict-autolabel.yml
Comment (teolemon) [reply .github/workflows/merge-conflict-autolabel.yml:1]: Ok with removing it. It's mostly useful on bigger repositories to stay on top and avoid PR getting too conflicted

**Review by VaiTon — APPROVED** [review]
